dependencies_main.g

```diff
@@ -0,0 +1,114 @@
+# This file contains some help functions making the parallel parts of the code 
+# in main_full_par_trafo.g more readable. (Dependency of main_full_par_trafo.g.)
+
+# Function in step 1
+ClearDownParameters := function(i, j, matrixC, TaskListClearDown, TaskListUpdateR, galoisField, dummyTask)
+	local list, C, D, number; # parameters for ClearDown as in subprograms.g
+
+	if (i = 1) and (j = 1) then
```

Comment:
As a further improvement:
It looks like the parameters only depend on `i` or on `j` independently but do not depend on `i` and `j`. This is easy to change for `C` and `D`.
For `list` I suggest to initialise it as `[dummyTask, dummyTask]` and `TaskListClearDown[i-1][j], TaskListUpdateR[i][j-1][j]` can be written to positions `[1]` and `[2]` independently of each other.

Having two `dummyTask` entries instead of one should not affect performance (I would be very surprised).

Reply:
I think the same could be done for the other functions in this file.

Reply:
Better yet, we can completely get rid of `dummyTask`. Just initialise `list := []`. `ScheduleTask` also accepts empty lists and lists with some unbound entries:
```
gap>  t1 := RunTask(x->x*x, 3);;
gap> ScheduleTask([,t1], x -> x, 5);
<protected object in shared region 0x9c85680 (id: 181360320)>
gap> TaskResult(last);              
5
gap> ScheduleTask([], x -> x, 5);;
gap> TaskResult(last);              
5
```